audbackend/core/backend.py

```diff
@@ -79,11 +87,11 @@ def checksum(
         """
         path = utils.check_path(path)
         version = utils.check_version(version)
+        path_with_version = self._path_with_version(path, version)
```

Comment:
I would move 
```python
path = utils.check_path(path)
version = utils.check_version(version)
```
into `self._path_with_version()` as it seems to be called always before `self._path_with_version()` is called.

This affects other methods in `Backend` as well.

Reply:
In addition, `utils.check_path()` can be removed in `latest_version()` as it is handled inside the call to `versions()` then.

Reply:
Yes, makes sense. I applied both suggestions.